include/bpf_helper_defs.h

```diff
@@ -419,6 +419,52 @@ EBPF_HELPER(int64_t, bpf_get_socket_cookie, (const void* ctx));
 #define bpf_get_socket_cookie ((bpf_get_socket_cookie_t)BPF_FUNC_get_socket_cookie)
 #endif
 
+/**
+ * @brief Copy a string into a buffer, of a fixed size.
+ *
+ * @param[in] dest Destination buffer.
+ * @param[in] dest_size Size of the destination buffer.
+ * @param[in] src Source string.
+ * @param[in] count Upper limit on bytes to copy.
+ *
+ * @retval 0 The operation was successful.
+ * @retval -ERANGE The destination buffer isn't large enough for the string being copied.
+ */
+EBPF_HELPER(int, bpf_strncpy_s, (char* dest, size_t dest_size, const char* src, size_t count));
+#ifndef __doxygen
+#define bpf_strncpy_s ((bpf_strncpy_s_t)BPF_FUNC_strncpy_s)
+#endif
+
+/**
+ * @brief Concatenate a string to an existing buffer, up to a certain number of characters.
+ *
+ * @param[in,out] dest Destination buffer.
+ * @param[in] dest_size Size of the destination buffer.
+ * @param[in] src String to append to the contents of dest.
+ * @param[in] count Upper limit on the characters to append to dest.
+ *
+ * @retval 0 Concatenation was successful.
+ * @returns Error code on failure, depending on the error.
```

Comment:
```suggestion
 * @retval<0 Error code on failure, depending on the error.
```
I'm not quite sure what doxygen does when you use both retval and returns.  All I know is that everywhere else we use one or the other rather than mixing them.  If you find that they work together can you add a comment with a pointer to a built out example?

Reply:
I'm happy to take your suggestion; I can't find a good example either way of mixing them, and that works well.